### ISSUE TITLE
Multi-tenancy support

### ISSUE THREAD

**Issue by roeltm** [matrix-org/synapse#11613]
**Description:**

For our organisation we want to use synapse to provide a chat service for all our customer companies. Since the organisations are not directly related we currently have to deploy a homeserver per customer. We would prefer to have it completely multi-tenant where we will run one homeserver with multiple domains, databases and other configurations.

Is this something that has already been discussed here? Would it be possible?

We might be interested in putting some of our developers on this, but is this something that would be desired? Before we spend all the time and find out we won't be able to get it merged.

**Comment by clokep** [comment]
I think this is essentially a duplicate of #8885. There's some chatter on there, but I don't think there's any concrete plan to support this.

I'd encourage you to chat with us (https://matrix.to/#/#synapse-dev:matrix.org) before starting any major work on this. I think there would be some interest, but it hasn't made it high on our priority list at the moment.

**Comment by callahad** [comment]
FWIW, we're very much on the fence about multitenancy. We considered it earlier in the year and concluded that it shouldn't be _that_ bad, but it would be quite invasive and we didn't yet have a clear need which was sufficiently motivating to justify that complexity.

We'd absolutely be interested in reviewing specific proposals for how you'd split things up, in advance of anyone knuckling down and doing the work.

We were especially concerned about growth and noisy neighbors: how do you quickly and reliably extract resource intensive sites from the multitenant host and move them to their own infrastructure?
